### PR TITLE
Prepare `polaris-for-vscode` for publish on marketplace

### PR DESCRIPTION
**Pull request by yurm04**
### WHY are these changes introduced?
Changes needed to get the Polaris for VSCode extension published on the VSCode marketplace https://github.com/Shopify/polaris/issues/5366

### WHAT is this pull request doing?
* adding Polaris icon
* updating package.json

### CODE REVIEW

File: polaris-for-vscode/package.json [hunk 1]
```diff
@@ -1,9 +1,22 @@
 {
   "name": "polaris-for-vscode",
   "displayName": "polaris-for-vscode",
-  "description": "Polaris Design System vscode tooling",
-  "repository": "https://github.com/Shopify/polaris-react",
+  "publisher": "Shopify",
+  "description": "Polaris Design System VSCode extension",
+  "icon": "polaris-for-vscode/docs/polaris-icon.png",
```
Comment (yurm04) [inline polaris-for-vscode/package.json]: no idea if this will resolve to the right URL 🤔 might need to use the resolved github url

Comment (charlespwd) [reply polaris-for-vscode/package.json]: It would be relative to the vsce package IIRC.

Comment (aaronccasanova) [reply polaris-for-vscode/package.json]: Yes, needs to be relative to the manifest (package.json). I took a similar approach with my VSCode extension: https://github.com/aaronccasanova/format-grid-areas/blob/12dc39d1836cd1ded6ab4ec02ad8064d17c2ce68/package.json#L14

Comment (aaronccasanova) [reply polaris-for-vscode/package.json]: Also, see the example manifest: https://code.visualstudio.com/api/references/extension-manifest#example

File: polaris-for-vscode/package.json [hunk 2]
```diff
@@ -1,9 +1,22 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/Shopify/polaris"
```
Comment (aaronccasanova) [inline polaris-for-vscode/package.json]:
```suggestion
    "url": "https://github.com/Shopify/polaris.git"
```
Not sure if this update is required, but it is in the [manifest example](https://code.visualstudio.com/api/references/extension-manifest#example).

File: polaris-for-vscode/package.json [hunk 3]
```diff
@@ -1,9 +1,22 @@
...
+  },
+  "homepage": "https://github.com/Shopify/polaris/tree/main/polaris-for-vscode",
```
Comment (aaronccasanova) [inline polaris-for-vscode/package.json]:
```suggestion
  "homepage": "https://github.com/Shopify/polaris/blob/main/polaris-for-vscode/README.md",
```
Same as above. Not sure if this update is required, but it matches the [manifest example](https://code.visualstudio.com/api/references/extension-manifest#example)

File: polaris-for-vscode/package.json (hunk 1 above)
Comment (aaronccasanova) [inline polaris-for-vscode/package.json]:
```suggestion
  "icon": "docs/polaris-icon.png",
```

**Review by romellogoodman — APPROVED** [review]

**Pull request merged**